From: Hein Roehrig <debian-bts@hein.roehrig.name>

Here is a patch to have the -localhost option in the x0vncserver
command as well... important when trying to secure the VNC connection
via ssh.

Thanks,
Hein

--- vnc4-4.0/unix/x0vncserver/x0vncserver.cxx.~1~	2004-10-23 10:59:42.000000000 -0600
+++ vnc4-4.0/unix/x0vncserver/x0vncserver.cxx	2004-10-23 11:48:41.000000000 -0600
@@ -48,6 +48,9 @@
 
 StringParameter displayname("display", "The X display", "");
 IntParameter rfbport("rfbport", "TCP port to listen for RFB protocol",5900);
+BoolParameter localhostOnly("localhost",
+                            "Only allow connections from localhost",
+                            false);
 IntParameter queryConnectTimeout("QueryConnectTimeout",
                                  "Number of seconds to show the Accept Connection dialog before "
                                  "rejecting the connection",
@@ -293,7 +296,7 @@
     QueryConnHandler qcHandler(dpy, &server);
     server.setQueryConnectionHandler(&qcHandler);
 
-    TcpListener listener((int)rfbport);
+    TcpListener listener((int)rfbport, localhostOnly);
     vlog.info("Listening on port %d", (int)rfbport);
 
     while (true) {




